### PR TITLE
GSOC: KubeVela HA Enhancement-Cluster Meta Backup&Restore

### PR DESCRIPTION
**Pull request by jefree-cat**
Signed-off-by: jefree-cat <jiahang_xu@cmbchina.com>

### CODE REVIEW

**Comment by jefree-cat** [comment]
add a project for kubevela

File: summerofcode/2023.md [hunk 1]
```diff
@@ -31,7 +31,8 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
   - [Eventing Sender Identity](#eventing-sender-identity)
   - [NetworkPolicy support for Knative Serving](#networkpolicy-support-for-knative-serving)
   - [Improving Observability in Knative Eventing](#improving-observability-in-knative-eventing)
-
+- [KubeVela](#kubevela)
```
Comment (nate-double-u) [inline summerofcode/2023.md:34]: Let's remove the ToC entry, we can add it at the end of the process.

File: summerofcode/2023.md [hunk 2]
```diff
@@ -73,3 +74,19 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+#### KubeVela HA Enhancement (VelaHA)
+
+- Description: KubeVela can be deployed in a single management cluster.The KubeVela management cluster has a single point of risk. When KubeVela management clusterdisaster occurs, business continuity cannot be guaranteed. In the production environment of enterprise use case(financial industry), KubeVela needs to have higher availability requirements. 
+Consider three enhancements to achieve this goal.
+* Cluster Meta backup & recovery,
+* Master redundancy,
+* Failover support
+- Expected outcome: velaha as addon for KubeVela with command line. cli> vela ha enable 
+- Recommended Skills: Golang, Kubernetes, KubeVela, CUELang
+- Mentor(s): Jiahang Xu @jefree-cat, Xiangbo Ma @fourierr
```
Comment (nate-double-u) [inline summerofcode/2023.md:89]: Could you provide email addresses for the mentors on this list? (either here, or you can DM me in slack)

**Review by nate-double-u — COMMENTED** [review]
Thanks for this @jefree-cat! A couple comments in line -- neither are blockers (I'll need to do a merge conflict update so if you're not able to remove the ToC items I can at that point).

File: summerofcode/2023.md (hunk 2 above)
Comment (fourierr) [reply summerofcode/2023.md:89]: ok

Comment (fourierr) [reply summerofcode/2023.md:89]: > Could you provide email addresses for the mentors on this list? (either here, or you can DM me in slack)

ok

**Comment by fourierr** [comment]
I have fixed conflict. Please help review@nate-double-u

**Comment by jefree-cat** [comment]
@nate-double-u  please review it .thx

File: summerofcode/2023.md [hunk 3]
```diff
@@ -129,6 +136,20 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
 - Difficulty: Hard
 - Upstream Issues (URL): https://github.com/kubevela/kubevela/issues/5366
 
+#### KubeVela HA Enhancement (VelaHA)
+
+- Description: KubeVela can be deployed in a single management cluster.The KubeVela management cluster has a single point of risk. When KubeVela management clusterdisaster occurs, business continuity cannot be guaranteed. In the production environment of enterprise use case(financial industry), KubeVela needs to have higher availability requirements.
+  Consider three enhancements to achieve this goal.
+* Cluster Meta backup & recovery,
+* Master redundancy,
+* Failover support
```
Comment (aliok) [inline summerofcode/2023.md:145]:
```suggestion
- Description: KubeVela can be deployed in a single management cluster.The KubeVela management cluster has a single point of risk. When KubeVela management clusterdisaster occurs, business continuity cannot be guaranteed. In the production environment of enterprise use case(financial industry), KubeVela needs to have higher availability requirements.  Consider these three enhancements to achieve this goal: cluster Meta backup & recovery, master redundancy, failover support
```

Sorry for nit picking, but the formatting in the doc breaks.

File: summerofcode/2023.md [hunk 4]
```diff
@@ -129,6 +136,20 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Expected outcome: velaha as addon for KubeVela with command line. cli> vela ha enable
```
Comment (aliok) [inline summerofcode/2023.md:146]: Is this task about integrating an existing solution named `velaha` to KubeVela and making it available in KubeVela CLI `vela`? Or, is it actually about implementing the HA requirements?

I am a bit confised and I think potential mentees would be confused too.

> cluster meta backup & recovery, master redundancy, failover support

I don't know about KubeVela technically, but I would like to raise the awareness that a GSoC project needs to be finished in 350h. Would you like the mentee to target all of these? Or, if it is too much, just some items from these 3?

If it is too much for a single open source beginner (GSoC program's definition of mentees) to implement all of these, you might want to just focus on one, and give more details around how to implement that.

File: summerofcode/2023.md [hunk 5]
```diff
@@ -129,6 +136,20 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Recommended Skills: Golang, Kubernetes, KubeVela, CUELang
+- Mentor(s): Jiahang Xu @jefree-cat jiahang_xu@cmbchina.com, Xiangbo Ma @fourierr maxiangboo@gmail.com
```
Comment (aliok) [inline summerofcode/2023.md:148]: 👍 , mentor(s) are in https://github.com/cncf/foundation/blob/main/project-maintainers.csv#L876

**Review by nate-double-u — DISMISSED** [review]
Hi @jefree-cat, could you update based on @aliok's feedback, and while you're making that update could you remove the table of contents stuff?

**Comment by jefree-cat** [comment]
yes,  vela ha issue is umbrella issue including meta backup and restore, failover issue. the gsoc issue just involves with meta scheduled backup and restore. there is a existing solution that velero as kubevela addon to integrate with vela command line.
I will give more details in issue link and reopen the pr. thanks. 
 @nate-double-u @aliok 

> Hi @jefree-cat, could you update based on @aliok's feedback, and while you're making that update could you remove the table of contents stuff?

**Comment by jefree-cat** [comment]
@nate-double-u @aliok already adjust the issue overall. please review. thx.

File: summerofcode/2023.md [hunk 6]
```diff
@@ -154,6 +154,19 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
 - Difficulty: Medium
 - Upstream Issues (URL): https://github.com/kubevela/kubevela/issues/5426
 
+#### KubeVela HA Enhancement-Cluster Meta Backup&Restore
+
+- Description: KubeVela can be deployed in a single management cluster.The KubeVela management cluster has a single point of risk. When KubeVela management clusterdisaster occurs, business continuity cannot be guaranteed. In the production environment of enterprise use case(financial industry), KubeVela needs to have higher availability requirements.  Consider these three enhancements to achieve this goal: cluster meta backup & recovery, master redundancy, failover support. Cluster meta backup & recovery is baseline feature for velaha.
+- Expected outcomes:
+  - implement velero as KubeVela addon to install velero in kubevela cluster.
+  - implement cli> vela ha enable to enable cluster meta backup with config scheduled(cron).
+  - implement cli> vela ha restore to restore cluster meta.
```
Comment (aliok) [inline summerofcode/2023.md:163]:
```suggestion
  - implement velero as KubeVela addon to install velero in kubevela cluster.
  - implement `cli> vela ha enable` to enable cluster meta backup with config scheduled(cron).
  - implement `cli> vela ha restore` to restore cluster meta.
```

**Review by aliok — APPROVED** [review]
/lgtm

Added a comment with a formatting suggestion. I would be happy to see that is included as well. But other than that, LGTM.

**Comment by jefree-cat** [comment]
> /lgtm
> 
> Added a comment with a formatting suggestion. I would be happy to see that is included as well. But other than that, LGTM.
thanks for your review feedback.

**Comment by jefree-cat** [comment]
@nate-double-u any other comments?

**Comment by aliok** [comment]
Thanks, LGTM!

**Review by nate-double-u — APPROVED** [review]
Thanks for this @jefree-cat, @fourierr, and @aliok for your reviews!

**Pull request merged**